google/cloud/storage/blob.py

```diff
@@ -2934,35 +2934,35 @@ def compose(
 
         :type if_generation_match: long
         :param if_generation_match:
-            (Optional) Make the operation conditional on whether the blob's
-            current generation matches the given value.  Setting to 0 makes the
-            operation succeed only if there are no live versions of the blob.
+            (Optional) Makes the operation conditional on whether the
+            destination object's current generation matches the given value.
+            Setting to 0 makes the operation succeed only if there are no live
+            versions of the object.
+
+            If a list of long is passed in, makes the operation conditional on whether the
+            current generation of each source blob matches the corresponding generation.
+            The list must match ``sources`` item-to-item.
+            (Deprecated: type(list of long) is supported for backwards-compatability reasons only.
+            Please use if_source_generation_match instead to match source objects generations.)
```

Comment:
Ouch!  PR #464 is also mangling the docstrings for the conditional parameters.

Reply:
Thanks for catching this. I think this is a special case where we will need to have a full explanation in the docstrings considering deprecation and backwards compatibility. Let me know if there is a better way to handle the docstrings here.

Reply:
I think I'll hold off merging #464 until this lands, and then look at how to merge this change.